ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/msck/MsckAnalyzer.java

```diff
@@ -63,13 +67,24 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     }
 
     Table table = getTable(tableName);
-    List<Map<String, String>> specs = getPartitionSpecs(table, root);
+    Map<Integer, List<ExprNodeGenericFuncDesc>> partitionSpecs = getFullPartitionSpecs(root, table, conf, false);
+    byte[] filterExp = null;
+    if (partitionSpecs != null & !partitionSpecs.isEmpty()) {
+      // explicitly set expression proxy class to PartitionExpressionForMetastore since we intend to use the
+      // filterPartitionsByExpr of PartitionExpressionForMetastore for partition pruning down the line.
+      conf.set(MetastoreConf.ConfVars.EXPRESSION_PROXY_CLASS.getVarname(),
```

Comment:
I don't think this will work - this is the ql module ; while `EXPRESSION_PROXY_CLASS` is a metastore conf key; in a remote metastore setup this set will probably have no effect...
have you tried it?
I think making a check and returning with an error that this feature is not available due to required conf change is fine

Reply:
Based on the implementation, I don't think there was never such a use case. I guess we always except it to be contatant
Refer:
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/msck/MsckAnalyzer.java#L66
https://github.com/apache/hive/blob/6f9c7bc490e31ee23556a316f2ac4a886d4eff1e/ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java#L1527
https://github.com/apache/hive/blob/6f9c7bc490e31ee23556a316f2ac4a886d4eff1e/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java#L259

Reply:
I think it would be great to throw an exception here; I think you should throw the type `SemanticException`
Throwing exceptions from in the `Analyzer` classes works nicely! :)

Reply:
Done!